### DISCUSSION TITLE
Allow greater customization of PDF export

### DISCUSSION THREAD

**Discussion by ahcsco** [GitbookIO/community#64] [Feature Requests]
### What's your feature/idea?

Today PDF export options in gitbook is very basic. Allow for greater customization like adding headers footers, font / color styling, cover page design. 

### What's your desired outcome?

Have better pdf export than the current bland version

### What's the impact of this for you?

- [ ] Critical
- [x] High
- [ ] Medium
- [ ] Low

**Comment by taszap** [comment]
This issue is linked to #90

**Comment by De-Ma** [comment]
Hi,

we just changed over to Gitbook and one of the major reason was the PDF export feature. We have customers who benefit from PDFs, and in our trail of Gitbook it looked promising. 
Now we run in to more and more problems with the PDF export and we are hoping for features / support:
- "Cards" are not exported correctly in PDFs
   - The "Cards" don't seam to export well at all 
- Site links are nowhere to be seen (when they are linked as page)
   - Links to internal spaces (when included as page) should also be included as full printout in the PDF export (or toggle if they should be included). We would like to use this to append our Copyright and Disclaimer page.
- No Table of content in exports
   - Expected behavior: Included table of content as Bookmarks, for easy PDF navigation
- Missing Footer/Title bar (or export date in the Footer)
   - What we would like: Customizable Header/Footer, maybe dependent on page style
- Missing Branding in exported PDFs
   - We would like to have our Logo and color theme represented in the export

**Comment by OtherToday** [comment]
We need this too. I am using it for student work and need a pdf back up for submission however lots of the formating is off and sometimes images don't get exported at all. It is not clear what rules govern page breaks. It would be handy if there was some tool that specified page breaks.

**Comment by azisso** [comment]
The basic requirement of having our branding is crucial. While there are many important ideas here, this one thing is a quick win and it's weird the logo goes away when you export to PDF - this will make them look much more professional

**Comment by a-asharapova** [comment]
We also need this. Simple branding and a table of contents are must-have for PDF documentation.

**Comment by outragedhuman** [comment]
We would also like to see this!

**Comment by sarahf16** [comment]
This would be really useful for us too.

**Comment by brunommst** [comment]
This would be very helpful for my team as well.

**Comment by pfj-joshua-golden** [comment]
Any updates on this? Our team uses Gitbook (Enterprise) and the ability to export to PDF and share some of our documents with other teams in our organization. The current PDF export is not professional in appearance and we're having to bend over backwards to move document content to another solution in order to provide indirect access.

**Comment by kfc2023** [comment]
exporting the whole space, too. decision makers don't want to peruse a portal.

**Reply by JaniceManwiller** [reply]
This is possible, although not very obvious - I only found out when I asked about it when they did the initial big UI update for published content.

When you click **Export as PDF** on a page, then on the "preview" page there is an **All Pages** button at the bottom left that exports the entire space.

**Comment by kfc2023** [comment]
but there is a limit of 100 pages. my doc is 300 pages and growing.

**Comment by irs-sesc1** [comment]
Some more PDF customization would be very helpful for us, as we need to provide offline documentation. Basics like a logo or header/footer texts would be more than enough.

**Comment by ykla** [comment]
The current export is almost unusable. There's no table of contents, image sizes can't be adjusted, and exporting long text is extremely complicated. My friend has written a tool for your developers to refer to: https://github.com/FreeBSD-Ask/gitbook-pdf-export

**Reply by ykla** [reply]
We now support export EPUB

**Comment by JaniceManwiller** [comment]
And I still cannot export the full documentation set - mine is currently 410 pages, but when I keep adding 50 pages, it errors out after 350.

**Comment by JFQueralt** [comment]
While some requests in this thread may take time to implement, I feel everyone would enjoy seeing some progress on this front.

I could do a summary of this tread if this comment gests 7 (half of the thread's participant count at this time) and get some items the devs could work on.

Meanwhile, I thing a basic feature such as "what you see on the published page is what you get on your PDF" is nothing controversial to ask for.
Especially cover images, which is very frustrating to not see reflected on the PDF.

**Comment by naleyjanhelge** [comment]
I love GitBook but PDF´s look kinda bad so I have had to manlually vopy the page content over to Google Docs and edit it there before sending PDF´s.

**Comment by DanielSmithGK** [comment]
Hi, this topic gets more and more attention also here.
We are providing important documents to customers and it just lacks professionalism with the current state the PDF export is in.

The following features would be extremely helpful (at least):

- Cover page (branding/logo or image option, title, display version, display date (Month Year))
- Table of contents
- Footer with a branding/logo-option

Is there any chance that this feature gets picked up with more priority?

We are also working with standard Gitbook editing features like images, tables, hints, code blocks, reusable content, cards, mermaid blocks and I would expect these also working in a PDF export.

I've seen very similar discussions here:

[Exporting to PDF in a clean format · GitbookIO · Discussion #1088](https://github.com/orgs/GitbookIO/discussions/1088)
[Retain branding in PDF export · GitbookIO · Discussion #1063](https://github.com/orgs/GitbookIO/discussions/1063)
[Display Table of contents in the exported PDF · GitbookIO · Discussion #1036](https://github.com/orgs/GitbookIO/discussions/1036)
[Feature request: Page numbers and table of contents in PDF export · GitbookIO · Discussion #1000](https://github.com/orgs/GitbookIO/discussions/1000)
[Request to export a PDF document to add a table of contents based on the summary · GitbookIO · Disc…](https://github.com/orgs/GitbookIO/discussions/669)
[PDF files with tables of content · GitbookIO · Discussion #534](https://github.com/orgs/GitbookIO/discussions/534)
[When using "export as PDF" some content does not appear or is cut off · GitbookIO · Discu…](https://github.com/orgs/GitbookIO/discussions/90)

Thanks and regards!
Daniel

**Reply by MissMogly** [reply]
I really hope gitbook can respond to this. the lack of TOC in PDF printing is real show stopper for my organization, branding is the second.

**Comment by kfc2023** [comment]
indeed. right now, i'm copying/pasting content into another app when i need
a pdf.

**Comment by naleyjanhelge** [comment]
I’d like to add a more detailed clarification of the issues we’re experiencing with the current PDF export:

- When clicking **Export to PDF**, GitBook shows a preview, but there is no “Save” button – only a Print button.  
  If we use the print option and select “Print to PDF,” the resulting PDF looks different from the preview GitBook shows. It also varies depending on which browser is used.

- There is **no simple way to apply branding**, such as adding a logo to the exported PDF.

- There is **no support for tables of contents** or contextual elements in the exported file.

- There is **no option for a cover page**, which is important for professional documentation.

- There is **no way to add custom headers/footers**, page headers, or page footers to the PDF export or print version.

- The biggest pain point is that the export lacks **proper page breaks**, causing content to break awkwardly between pages.  
  Layout and formatting often look strange in the final PDF or printed output.

This issue goes back to early 2023, and since PDF export is a fundamental feature for many of us, we would really appreciate if this part of the platform could receive some dedicated attention. We truly appreciate GitBook, but reliable and consistent PDF export is critical for our workflow.

**Comment by dt-dmitry-kuzminsky** [comment]
Hey team, +1 to this request. We are really missing out on being able to share a professional-looking PDF with company branded cover page.
At least allow the option to add a cover page.
Alternatively, expose the PDF export methods so that we could code our own solution as an integration
